logstash-core/lib/logstash/modules/settings_merger.rb

```diff
@@ -22,12 +22,19 @@ module LogStash module Modules module SettingsMerger
   extend self
 
   def merge(cli_settings, yml_settings)
+    unless cli_settings.is_a?(LogStash::Util::ModulesSettingArray)
+      raise IllegalArgumentException.new "cli_settings must be of type LogStash::Util::ModulesSettingArray"
```

Comment:
:safety_pin: too much Ruby<->Java? :wink: 
there's a `java_import java.lang.IllegalArgumentException` -> gets `IllegalArgumentException` into the global scope, but I believe in general we should avoid raising a Java exception from Ruby unless the consumer is Java.

Reply:
the string type checking here generate an error in startup because during Logstash bootstrap the variables are set as https://github.com/elastic/logstash/blob/7db4692044213f1b5d464712b1d73400be143583/logstash-core/lib/logstash/environment.rb#L42 that's by default a `RubyArray` instance, later coerced to `ModulesSettingArray`